src/mqt/ionshuttler/_json_utils.py

```
@@ -0,0 +1,207 @@
+    from collections.abc import Mapping
+
+
+def require_mapping(data: object, label: str) -> dict[str, object]:
+    """Return a JSON object.
+
+    Raises:
+        ValueError: If the value is not a JSON object.
+    """
+    if not isinstance(data, dict):
+        msg = f"{label} must be a JSON object"
+        raise ValueError(msg)  # ruff: ignore[type-check-without-type-error] - JSON format errors use ValueError.
```

Ruff (0.16.1)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)